File: backend/src/main/java/es/udc/tfgproject/backend/model/services/BusinessCatalogService.java
package es.udc.tfgproject.backend.model.services;

import es.udc.tfgproject.backend.model.entities.CompanyAddress;

public interface BusinessCatalogService {

	Block<CompanyAddress> findCompanies(Long companyCategoryId, Long cityId, String street, String keywords, int page,
			int size);

	Block<CompanyAddress> findAllCompanies(String keywords, int page, int size);

}

File: frontend/src/backend/businessCatalogService.js
import {config, appFetch} from './appFetch';

export const findCompanies = ({companyCategoryId, cityId, street, keywords, page}, 
    onSuccess) => {

    let path = `/businessCatalog/companies?page=${page}`;

    path += companyCategoryId ? `&companyCategoryId=${companyCategoryId}` : "";
    path += cityId ? `&cityId=${cityId}` : "";
    path += street.length > 0 ? `&street=${street}` : "";
    path += keywords.length > 0 ? `&keywords=${keywords}` : "";

    appFetch(path, config('GET'), onSuccess);

}

export const findAllCompanies = ({keywords, page}, 
    onSuccess) => {

    let path = `/businessCatalog/companies/findAllCompanies?page=${page}`;

    path += keywords.length > 0 ? `&keywords=${keywords}` : "";

    appFetch(path, config('GET'), onSuccess);

}
